rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/ValidatedRpkiObjects.java

```diff
@@ -291,4 +258,70 @@ private void notifyListeners() {
         Locks.locked(dataLock.readLock(), () ->
             listeners.forEach(listener -> listener.accept(validatedObjectsByTrustAnchor.values())));
     }
+
+    public static class Accumulator {
+        private final TrustAnchorData trustAnchorData;
+        private final List<Key> validatedObjectKeys = Collections.synchronizedList(new ArrayList<>());
```

Comment:
I get a bit skittish when i see Collections.synchronizedList`. Since it seems like we only need `size` for logging (and could return it from `markReachable`): Can we use a wait-free datastructure/`ConcurrentLinkedQueue` here?

Reply:
Synchronized list was used here because it is the data structure currently used and we use `parallelStream` in the validation process. A cleaner approach would be to stream the results back and use `flatMap` to collect all data from the various threads. Not sure if it matters much for performance though.